connector/connect/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -253,6 +254,23 @@ message Repartition {
   bool shuffle = 3;
 }
 
+// Compose the string representing rows for output.
+// It will invoke 'Dataset.showString' to compute the results.
+message ShowString {
+  // (Required). The input relation.
+  Relation input = 1;
+
+  // (Required) Number of rows to show.
```

Comment:
Optional?

Reply:
I think we should have a proto message style guide soon that people agree.

For example required field with `optional` notion is very confusing.

IMO required field does not need `optional` to know set/unset. We say it is required which means clients should always offer a value. Server side just use the value.

Reply:
Or we say if a field is required, we use `optional` to offer the `hasXXX` method. This is also fine, but probably we need a consensus. I will refine what I had in https://github.com/apache/spark/pull/38605.

Reply:
I prefer to make it `required` here.
we can not invoke the `show` methods since they just print and return Unit.
to get a string, we need to invoke `showString` which requires all the parameters.

Reply:
is there a best practice in the industry? making everything optional is also fine to me, so that the server side can detect more malformed proto messages.